core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,1197 @@
+
+(exp_models,) = models.Registry.import_models([models.NAMES.exploration])
+
+
+class ExpStateValidationJob(base_jobs.JobBase):
+    """Job that tests the general state, rte and interaction validation"""
+
+    @staticmethod
+    def filter_invalid_state_rte_values(
+        states_dict: dict[str, state_domain.State]
+    ) -> List[Dict[str, List[Dict[str, List[str]]]]]:
+        """Returns the errored state RTE values, validates the following -
+
+        - Image tags contain filepath, alt, and caption attributes,
+        where caption can be an empty string with at most 160
+        characters and alt should have at least 5 characters.
+        - Math tags contain math_content, raw_latex, and svg_filename
+        attributes, where svg_filename has an SVG extension.
+        - Skillreview tags contain text attributes, text is non-empty.
+        - Video tags contain video_id, start, end, and autoplay attributes,
+        where start is before end.
+        - Link tags contain text where text is non-empty.
+
+        Args:
+            states_dict: dict[str, state_domain.State]. The dictionary
+                containing state name as key and State object as value.
+
+        Returns:
+            states_with_values: List[Dict[str, List[Dict[str, List[str]]]]].
+            The list of dictionaries containing the errored values.
+        """
+        states_with_values = []
+        for key, value in states_dict.items():
+            rte_components_errors = []
+            rte_components = html_cleaner.get_rte_components(
+                value.content.html)
+            for rte_component in rte_components:
+                # RTE image validations for caption, alt and filepath.
+                if rte_component['id'] == 'oppia-noninteractive-image':
+                    # Validates if caption value is <= 160.
+                    cap_with_value = rte_component['customization_args'][
+                        'caption-with-value']
+                    alt_with_value = rte_component['customization_args'][
+                        'alt-with-value']
+                    file_with_value = rte_component['customization_args'][
+                        'filepath-with-value']
+
+                    if len(cap_with_value) > 160:
+                        rte_components_errors.append(
+                            f'State - {key} Image tag caption value '
+                            f'is greater than 160 '
+                            f'having value {file_with_value}.'
+                        )
+
+                    # Validates if alt value is greater than 5.
+                    if len(alt_with_value) < 5:
+                        rte_components_errors.append(
+                            f'State - {key} Image tag alt value '
+                            f'is less than 5 '
+                            f'having value {file_with_value}.'
+                        )
+
+                    # Validates if filepath extension is svg.
```

Comment:
Wait, images in explorations can only be SVG images?

Reply:
Yes, This check is mentioned [here](https://docs.google.com/spreadsheets/d/1kNLF-UkS_Qk0igNCGwLjJXM0ByHalqli41GPvWF43PE/edit#gid=259309225) on the spreadsheet.

Reply:
I don't see that in the sheet

Reply:
Oh okay so currently the `svg diagram` and the `android image` row are combined, previously they were separated in the RTE but now they are combined, I had a discussion with Sean as I wasn't able to find any options related to `svg diagram` in the RTE he mentioned that it is now the part of the `android image` section and whole combine as `oppia-noninteractive-image`
Thanks

Reply:
Okay, and what does that mean? Should it always be SVG or not?

Reply:
Yes it should always be `SVG`